### PR TITLE
EOS-23614 : Code changes for s3 single entry script with call to s3authserver script

### PR DESCRIPTION
**Pull request by shahidSTX**
Signed-off-by: Mohammed Shahid <mohammed.shahid@seagate.com>

# Problem Statement
- Create a single entry script s3_start to support argument: --service [ s3server | s3authserver | s3bgproducer | s3bgconsumer ]

# Design
-  Create basic structure for allowing all service provisioning separately.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: scripts/provisioning/s3_start [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  try:
+    if args.service == 's3server':
+      # Action
+    elif args.service == 's3authserver':
+      os.system("sh /opt/seagate/cortx/auth/startauth.sh")
+    elif args.service == 's3bgproducer':
+      # Action
+    elif args.service == 's3bgconsumer':
+      # Action
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_start:57]: add 'else' here to handle wrong/unsupported argument (or define specific accepted list in the arg parser).

File: scripts/provisioning/s3_start [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    logger.addHandler(chandler)
+
+    return logger
+
+def create_logger_directory():
+    """Create log directory if not exsists."""
+    _logger_directory = os.path.join(s3_single_entry_log_directory)
+    if not os.path.isdir(_logger_directory):
+        try:
+            os.mkdir(_logger_directory)
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_start:102]: This will fail if  parent directories don't exist.  Use `os.makedirs()`?...

**Review by t7ko-seagate — CHANGES_REQUESTED** [review]

File: scripts/provisioning/s3_start (hunk 1 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_start]: Done.

File: scripts/provisioning/s3_start (hunk 2 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_start]: Done.

**Review by t7ko-seagate — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

**Comment by shahidSTX** [comment]
mini-provisioning was passing earlier, now it's failing as below :
image
Re-triggering fails with same above error.

**Comment by shahidSTX** [comment]
Update from Nilesh : infra changes are in progress and that is causing mini-provisioning failure.

**Comment by vimalraghwani20** [comment]
run mini-provisioning

**Comment by vimalraghwani20** [comment]
run vm-deployment

**Comment by vimalraghwani20** [comment]
run premerge-test

**Comment by gauravchaudhari02** [comment]
run premerge-test

**Comment by vimalraghwani20** [comment]
run mini-provisioning

**Comment by vimalraghwani20** [comment]
run premerge-test

**Comment by vimalraghwani20** [comment]
run vm-deployment

**Comment by vimalraghwani20** [comment]
run vm-deployment

**Comment by shahidSTX** [comment]
All jobs were passing except VM deployment 7.9 which was failing but not in s3, so this PR is ready for merge.

**Comment by vimalraghwani20** [comment]
run premerge-test

**Review by nileshgovande — APPROVED** [review]

**Pull request merged**